play-media-native/scripts/publishAPKGitHub.mts

```diff
@@ -15,44 +16,50 @@ const defaultHeaders = {
   'X-GitHub-Api-Version': '2022-11-28',
 };
 
+const getBuildNumber = (tag: string): number =>
+  tag
+    .split('-')[2]
+    .substring(1)
+    .split('.')
+    .map((x: string) => parseInt(x))[2];
+
 const updateTagVersion = (tag: string): string => {
-  let tagVersion = tag.split('-')[2];
-  tagVersion = tagVersion.substring(1);
-
-  let tagVersionParts = tagVersion.split('.').map((x) => parseInt(x));
-
-  if (tagVersionParts[2] < 9) {
-    tagVersionParts[2] += 1;
-  } else if (tagVersionParts[1] < 9) {
-    tagVersionParts[1] += 1;
-  } else {
-    tagVersionParts[0] += 1;
-  }
-  return tagVersionParts.join('.');
+  let buildNumber = getBuildNumber(tag);
+  buildNumber += 1;
+
+  return `v${packageJson.baseVersion}.${buildNumber}`;
 };
 
 const updateTagName = (tag: string): string => {
   let newTagVersion = updateTagVersion(tag);
 
-  return sourceBranch === 'main' ? `APK-stable-v${newTagVersion}` : `APK-nightly-v${newTagVersion}`;
+  return sourceBranch === 'main' ? `APK-stable-${newTagVersion}` : `APK-nightly-${newTagVersion}`;
 };
 
-// Retrieve the latest relevant tag name and update it
-let tagName = '';
-if (sourceBranch === 'develop') {
-  tagName = await request('GET /repos/{owner}/{repo}/releases', {
-    ...defaultOptions,
-    headers: defaultHeaders,
-  }).then(
-    (res: { data: any[] }) =>
-      res.data.find((release: { prerelease: boolean }) => release.prerelease === true).tag_name
-  );
-} else {
-  tagName = await request('GET /repos/{owner}/{repo}/releases/latest', {
-    ...defaultOptions,
-    headers: defaultHeaders,
-  }).then((res: { data: { tag_name: string } }) => res.data.tag_name);
-}
+// Retrieve the latest nightly and stable tag name matching the base version and update the one with the highest build number
+const tagName = await request('GET /repos/{owner}/{repo}/releases', {
+  ...defaultOptions,
+  headers: defaultHeaders,
+}).then((res: { data: any[] }) => {
+  const latestNightlyTagName = res.data.find(
```

Comment:
I think this logic will not work if we bump the baseVersion minor or major version and we have no existing release with that base version. Both `latestNightlyTagName` and `latestStableTagName` will be null.

Also, if GitHub returns the releases in an order different from created date descending, we could get in trouble by returning the first one in the array that matches the baseVersion, even if it is not the highest build number.

I created Jest unit tests and improved the logic to have a single function that returns the tag with the absolute highest build number that matches the base version. I also return undefined when none exist. And I improved the logic that follow to handle the case when it is undefined.